lib/miq_pglogical/connection_handling.rb

```diff
@@ -2,10 +2,27 @@ class MiqPglogical
   module ConnectionHandling
     extend ActiveSupport::Concern
 
+    included do
+      delegate :logical_replication_supported?, :to => :class
+    end
+
     module ClassMethods
+      def logical_replication_supported?
+        return @logical_replication_supported if defined?(@logical_replication_supported)
+
+        is_superuser = ActiveRecord::Base.connection.select_value("select usesuper from pg_user where usename = CURRENT_USER;", "SQL")
+        warn "\e[33mWARNING: Current user is NOT a superuser, logical replication will not function.\e[0m" unless is_superuser
+
+        @logical_replication_supported = is_superuser
+      end
+
       def with_connection_error_handling
         retry_attempted ||= false
-        yield
+        if logical_replication_supported?
+          yield
+        else
+          # Silently do no harm since logical replication is not supported.
```

Comment:
@Fryguy @kbrock this is where the null object type behavior is occurring.  We don't need an actual object, just don't yield anything if logical replication is not supported.

Reply:
Note, this greatly decreases the guard clauses on this `supported` check.  There are a few remaining just because there are side effects that occur outside the block for `with_connection_error_handling`

Reply:
Should we explicitly return a nil here, just to make the code clearer?

Reply:
Good idea.